pkg/api/v1/machine.go

```diff
@@ -28,6 +29,7 @@ func NewMachineGroup(g *echo.Group, providerRepo repository.ProviderRepository,
 		config:       config,
 	}
 
+	g.GET("/:workspaceId/gpus", auth.WithWorkspaceAuth(group.GPUCounts))
```

Comment:
not sure if you mentioned this, but why do we need workspace Id here?

Reply:
> not sure if you mentioned this, but why do we need workspace Id here?

If I remember correctly, there was an issue with workspace auth if it was not included.